profiling/src/lib.rs

```diff
@@ -615,36 +621,44 @@ extern "C" fn rinit(r#type: c_int, module_number: c_int) -> ZendResult {
     #[cfg(feature = "allocation_profiling")]
     {
         if profiling_allocation_enabled {
-            if !is_zend_mm() {
-                // Neighboring custom memory handlers found
-                debug!("Found another extension using the ZendMM custom handler hook");
-                unsafe {
-                    zend::zend_mm_get_custom_handlers(
-                        zend::zend_mm_get_heap(),
-                        &mut PREV_CUSTOM_MM_ALLOC,
-                        &mut PREV_CUSTOM_MM_FREE,
-                        &mut PREV_CUSTOM_MM_REALLOC,
-                    );
-                }
-            }
-
-            unsafe {
-                zend::ddog_php_prof_zend_mm_set_custom_handlers(
-                    zend::zend_mm_get_heap(),
-                    Some(alloc_profiling_malloc),
-                    Some(alloc_profiling_free),
-                    Some(alloc_profiling_realloc),
-                );
-            }
-
-            if is_zend_mm() {
-                error!("Memory allocation profiling could not be enabled. Please feel free to fill an issue stating the PHP version and installed modules. Most likely the reason is your PHP binary was compiled with `ZEND_MM_CUSTOM` being disabled.");
+            if unsafe { zend::ddog_php_jit_enabled() } == true {
```

Comment:
Good idea 👍

Reply:
One thing that comes to my mind: AFAIK you could enable JIT on a per directory basis. So it could potentially happen that in the first request the configuration has JIT disable, while in the second request it is enabled (as it hit another directory and INI settings were loaded from `.user.ini` or `.htaccess` files).

Reply:
Found it:

Name | Standard | Veränderbar | Changelog
-- | -- | -- | --
opcache.jit | "tracing" | PHP_INI_ALL | Available as of PHP 8.0.0
opcache.jit_buffer_size | "0" | PHP_INI_SYSTEM | Available as of PHP 8.0.0

So the `jit_buffer_size` itself not, but one could enable/disable the JIT itself on a per directory basis.

Reply:
That's why we're looking at buffer_size, as that's a global value (and influenced by a PHP_INI_SYSTEM). You *can* disable the JIT at any time at runtime, if it's enabled (but not enable it if disabled).
Basically checking whether jit is enabled at all.
This is necessary in the tracers case as it does user opcode modifications which necessarily are global.

In this case, you could also check on a per-request basis, but then look at the `"on"` key instead of buffer_size.

Reply:
I like the way you are thinking 😉 
I commited a patch to only check once